### PR TITLE
Making a working release

### PR DESCRIPTION
**Pull request by squeedee**
release.yml got a bit messed up

Would like this merged so anyone following the Install instructions in README.md are succesful.

### CODE REVIEW

File: releases/release.yaml [hunk 1]
```diff
@@ -776,44 +777,44 @@ metadata:
...
+          readOnly: true
       serviceAccount: cartographer-controller
       volumes:
-        - name: cert
-          secret:
-            defaultMode: 420
-            secretName: cartographer-webhook
-      containers:
-        - name: cartographer-controller
-          image: 192.168.0.108:5000/controller/cmd-72b2fdd4563b5da9ab700e504fe28dcd@sha256:9321488b0f837dfd6ecb3511aced50cc3691f1601e3b8da0d156ecf0e2001d79
```
Comment (cirocosta) [inline releases/release.yaml:796]: uuuf 😅

Comment (cirocosta) [reply releases/release.yaml:796]: actually, it made me think that maybe for _"create release and upload files to release #7_" we might want to run e2e against the release that has been generated _before_ publishing it 🤔  it'd certainly catch this (aside from being a good validation anyway)

**Review by cirocosta — APPROVED** [review]
lgtm

to verify that it's indeed all good:

1. go to the branch where e2e testing was added

```bash
git checkout 181/e2e-test
```

2. disable the auto-generation of a release

```diff
diff --git a/hack/ci/e2e.sh b/hack/ci/e2e.sh
index 1d33c71..d860097 100755
--- a/hack/ci/e2e.sh
+++ b/hack/ci/e2e.sh
@@ -35,7 +35,7 @@ main() {
                 start_registry
                 start_local_cluster

-                generate_cartographer_release
+                # generate_cartographer_release
                 install_dependencies

                 setup_example
```

3. the grab the release from this pr


```bash
git checkout 769ca79b2464d5c149356884b82ed1d7954b3ac3 ./releases/release.yaml
```

4. run the tests

```
./hack/ci/e2e.sh
```

result:

```1:01:18PM: ---- applying 1 changes [9/10 done] ----
1:01:18PM: create workload/dev (carto.run/v1alpha1) namespace: default
1:01:18PM: ---- waiting on 1 changes [9/10 done] ----
1:01:18PM: ok: reconcile workload/dev (carto.run/v1alpha1) namespace: default
1:01:18PM: ---- applying complete [10/10 done] ----
1:01:18PM: ---- waiting complete [10/10 done] ----

Succeeded

        testing

- attempt 15
- attempt 14
- attempt 13
- attempt 12
- attempt 11
- attempt 10

        SUCCEEDED! sweet

```

File: releases/release.yaml (hunk 1 above)
Comment (squeedee) [reply releases/release.yaml:796]: @pivotal-todd-ritchie I think this is worth considering. It does not have to happen now of course, could be a story for later.

**Pull request merged**